From 37037b2ac949d73746195c8ce9c13e6d28ff5b86 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ga=C3=ABl=20Bonithon?= <gael@xfce.org>
Date: Tue, 18 Jun 2024 18:56:35 +0200
Subject: [PATCH] build: Fix LDFLAGS

* Align panel-plugin LDFLAGS with other plugins
* Remove flags from lib that break build when lld is used as linker

Fixes: #48
---
Patch-Source: https://gitlab.xfce.org/panel-plugins/xfce4-sensors-plugin/-/commit/37037b2ac949d73746195c8ce9c13e6d28ff5b86

 lib/Makefile.am          | 2 --
 panel-plugin/Makefile.am | 3 +--
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/lib/Makefile.am b/lib/Makefile.am
index 2789b5a5..84f2dccf 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -51,8 +51,6 @@ endif
 libxfce4sensors_la_CXXFLAGS = $(libxfce4sensors_la_CFLAGS)
 
 libxfce4sensors_la_LDFLAGS = \
-	-export-dynamic \
-	-export-symbols-regex "^[^_].*" \
 	-version-info $(LIBXFCE4SENSORS_VERINFO)
 
 # @PLATFORM_LDFLAGS@
diff --git a/panel-plugin/Makefile.am b/panel-plugin/Makefile.am
index 88c8608f..538bc92a 100644
--- a/panel-plugin/Makefile.am
+++ b/panel-plugin/Makefile.am
@@ -34,8 +34,7 @@ libxfce4_sensors_plugin_la_LDFLAGS = \
 	-avoid-version \
 	-module \
 	-no-undefined \
-	-export-dynamic \
-	-export-symbols-regex "^[^_].*" \
+	-export-symbols-regex '^xfce_panel_module_(preinit|init|construct)' \
 	$(PLATFORM_LDFLAGS)
 
 # .desktop file
-- 
GitLab

